### PR TITLE
Added Unit Tests for tasks and Integration Tests

### PR DESCRIPTION
**Pull request by marcin-kulik**
- added unit tests for tasks
- changed One-to-many relationship to unidirectional One-to-many
- added integration tests

### CODE REVIEW

File: src/it/java/rest/demo/ToDoControllerIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    private ToDoRepository toDoRepository;
+
+    private ToDo toDo1;
+    private List<ToDo> toDos;
+    private List<Task> tasks;
+    private LocalDate date = LocalDate.now();
+    private DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+
+    @Test
+    @DisplayName("When GET and no ToDos, Then NOT_FOUND")
```
Comment (boseabhishek) [inline src/it/java/rest/demo/ToDoControllerIntegrationTest.java]: Can it be written in a more meaningful way?


 Same for all other tests?

File: src/test/java/rest/demo/controller/ToDoControllerTest.java [hunk 2]
```diff
@@ -56,41 +54,75 @@ void getToDos_Accepted() {
     }
 
     @Test
-    @DisplayName("when ToDos do not exist in repository, then we get NOT_FOUND")
+    @DisplayName("When no ToDos, Then NOT_FOUND")
```
Comment (boseabhishek) [inline src/test/java/rest/demo/controller/ToDoControllerTest.java]: when no ToDo retrieved/received, return NOT_FOUND

Comment (boseabhishek) [reply src/test/java/rest/demo/controller/ToDoControllerTest.java]: maybe same for other tests

File: src/it/java/rest/demo/ToDoControllerIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+    @Test
+    @DisplayName("When POST with ToDos, Then ToDos")
+    public void whenPostingToDos_ThenSuccessfulAndReturnsToDos() throws Exception {
+
+        createToDos();
+        createTasks();
+        toDo1.setTasks(tasks);
+
+        Gson gson = new Gson();
```
Comment (boseabhishek) [inline src/it/java/rest/demo/ToDoControllerIntegrationTest.java]: Can gson be not used in actual controllers as well, replacing Jackson

File: pom.xml [hunk 4]
```diff
@@ -77,6 +81,36 @@
 				<groupId>org.springframework.boot</groupId>
 				<artifactId>spring-boot-maven-plugin</artifactId>
 			</plugin>
+			<plugin>
```
Comment (boseabhishek) [inline pom.xml]: What does all these do?

**Review by boseabhishek — CHANGES_REQUESTED** [review]
Looks good - though I have put some revengeful comments :p

File: pom.xml (hunk 4 above)
Comment (marcin-kulik) [reply pom.xml]: If you mean the below bit, it makes Maven see the folder 'src/it/java' as a source folder for tests

<plugin>
				<groupId>org.codehaus.mojo</groupId>
				<artifactId>build-helper-maven-plugin</artifactId>
				<executions>
					<execution>
						<id>add-test-source</id>
						<phase>process-resources</phase>
						<goals>
							<goal>add-test-source</goal>
						</goals>
						<configuration>
							<sources>
								<source>src/it/java</source>
							</sources>
						</configuration>
					</execution>
				</executions>
			</plugin>
			<plugin>

File: src/it/java/rest/demo/ToDoControllerIntegrationTest.java (hunk 3 above)
Comment (marcin-kulik) [reply src/it/java/rest/demo/ToDoControllerIntegrationTest.java]: Done

File: src/it/java/rest/demo/ToDoControllerIntegrationTest.java (hunk 1 above)
Comment (marcin-kulik) [reply src/it/java/rest/demo/ToDoControllerIntegrationTest.java]: done

File: src/test/java/rest/demo/controller/ToDoControllerTest.java (hunk 2 above)
Comment (marcin-kulik) [reply src/test/java/rest/demo/controller/ToDoControllerTest.java]: done

**Comment by boseabhishek** [comment]
Looks cool.. merging

**Pull request merged**